### PR TITLE
Add 'index' attribute to frame objects

### PR DESCRIPTION
**Pull request by t-wissmann**
This equips the frame objects with a dynamic attribute 'index'
containing the frame's frame index. Since there is no documentation for
frame objects yet, there is no documentation for the 'index' attribute
here.

This feature has been requested by @bew during the discussion of #906.

### CODE REVIEW

File: tests/test_layout.py [hunk 1]
```diff
@@ -952,3 +952,23 @@ def test_split_something_in_between(hlwm, splitmode, context):
...
+        assert hlwm.get_attr(f'{object_path}.index') == index_prefix
+        if '0' in hlwm.list_children(object_path):
+            verify_frame_tree(f'{object_path}.0', f'{index_prefix}0')
+            verify_frame_tree(f'{object_path}.1', f'{index_prefix}1')
+
+    for index in split_index:
+        hlwm.call(['split', 'auto', '0.5', index])
+
+        # after each splitting operation, check that
+        # the frame_index attribute is correct:
```
Comment (blarz) [inline tests/test_layout.py:973]: It is just "index" not frame_index, right?

Comment (blarz) [reply tests/test_layout.py:973]: Is frame_index maybe a better name for the actual attribute? Not sure, seems redundant

Comment (t-wissmann) [reply tests/test_layout.py]: I've changed the comment to "the frame's index" attribute. I think to avoid redundancy, we should keep the name `index`.

Comment (blarz) [reply tests/test_layout.py]: Cool, thanks!

**Pull request merged**